chore(build): add missing dependencies and fix make path

quatrox opened the pull request:
This commit adds the following fixes to simple_make.sh
 - cmake -H -B is undocumented and not officially supported, so replace
   it with the standard way
   Fixes #4267
 - fix build via "opensuse" docker image
   + zypper in sudo (if not installed)
   + replace external program "which" with shell built-in "command -v"
   + update and sorte depencies for OpenSUSE
   + improve the distro test, as OpenSUSE may have apt installed, but
     not necessarily lsb_release.

Note:
OpenSUSE has no package called ffmpeg-devel in the official repos, but it
it has ffmpeg2-devel.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/4273)
<!-- Reviewable:end -->

quatrox committed:
chore(build): add missing dependencies and fix make path

This commit adds the following fixes to simple_make.sh
 - cmake -H -B is undocumented and not officially supported, so replace
   it with the standard way
   Fixes #4267
 - fix build via "opensuse" docker image
   + zypper in sudo (if not installed)
   + replace external program "which" with shell built-in "command -v"
   + update and sorte depencies for OpenSUSE
   + improve the distro test, as OpenSUSE may have apt installed, but
     not necessarily lsb_release.

Note:
OpenSUSE has no package called ffmpeg-devel in the official repos, but it
it has ffmpeg2-devel.

quatrox commented:
This replaces #4270 and #4273

sudden6 commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> tested on openSUSE 42.2

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4273#-:-KfjqVrxPbwK8NRSiCFH:bojcwkf)*
<!-- Sent from Reviewable.io -->

zetok commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[simple_make.sh, line 122 at r1](https://reviewable.io:443/reviews/qtox/qtox/4273#-KfkCkOiwub1wm7y4QnV:-KfkCkOiwub1wm7y4QnW:bbpyqpc) ([raw file](https://github.com/qtox/qtox/blob/9b3d919d27a9d3bbd85f02c77cecdcf8db817dd4/simple_make.sh#L122)):*
> ```Shell
>         automake
>         cmake
>         ffmpeg2-devel
> ```

`ffmpeg-devel` ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4273)*
<!-- Sent from Reviewable.io -->

quatrox commented:
@zetok there are no package called "ffmpeg-devel" or "libffmpeg-devel" in the official repos for OpenSUSE Tumbleweed:


S | Name                     | Summary                                     | Type      
--+--------------------------+---------------------------------------------+-----------
  | ffmpeg                   | Library for working with various multimed-> | package   
  | ffmpeg                   | Library for working with various multimed-> | srcpackage
  | ffmpeg-debuginfo         | Debug information for package ffmpeg        | package   
  | ffmpeg-debugsource       | Debug sources for package ffmpeg            | package   
  | ffmpeg2                  | Library for working with various multimed-> | srcpackage
  | ffmpeg2-debugsource      | Debug sources for package ffmpeg2           | package   
i | ffmpeg2-devel            | Development files for FFmpeg 2.x libraries  | package   
  | ffmpegthumbs             | FFmpeg-based thumbnail creator for video -> | package   
  | ffmpegthumbs             | FFmpeg-based thumbnail creator for video -> | srcpackage
  | ffmpegthumbs-debuginfo   | Debug information for package ffmpegthumbs  | package   
  | ffmpegthumbs-debugsource | Debug sources for package ffmpegthumbs      | package

sudden6 commented:
<img class="emoji" title=":lgtm_cancel:" alt=":lgtm_cancel:" align="absmiddle" src="https://reviewable.io/lgtm_cancel.png" height="20" width="61"/> 

I think the correct way to include ffmpeg headers is to include  the needed `libav*-devel packages` because these are the headers for ffmpeg 3.x series

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4273#-:-Kfkb2zh8O-qCf_mLr8r:b-18zbjc)*
<!-- Sent from Reviewable.io -->

quatrox committed:
chore(build): switch from ffmpeg2-devel to libav*-devel for OpenSUSE

Switched to libav*-devel because ffmpeg-devel is not available on
OpenSUSE in the official repos, and libav*-devel provides the headers
for FFmpeg 3.x series.

zetok commented:
>cmake -H -B is undocumented and not officially supported, so replace
it with the standard way

NB: `-B` is documented, at least as far as `cmake --help-full` goes.

Anyway, <img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4273#-:-KfpKkuGKVHqmiRjU1RS:bdr60fw)*
<!-- Sent from Reviewable.io -->

sudden6 commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4273#-:-KfpOC0VwPrJ7NWceugH:b-jbsf8z)*
<!-- Sent from Reviewable.io -->

sudden6 commented:
@quatrox would you also be interested in updating https://github.com/qTox/qTox/blob/master/INSTALL.md for openSUSE in another PR?
